pkg/plugins/manager.go

```diff
@@ -0,0 +1,113 @@
+package plugins
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+var (
+	ErrPluginOfWrongType    = errors.New("plugin of the wrong type")
+	ErrPluginNameNotFound   = errors.New("unknown plugin name")
+	ErrUninitializedManager = errors.New("uninitialized plugins manager")
+)
+
+var allowedProtocols = []plugin.Protocol{
+	plugin.ProtocolGRPC,
+}
+
+// PluginIdentity identifies the plugin's version and executable location.
+type PluginIdentity struct {
+	Version int
+	// Cmd is the command that is used to run the plugin executable on the local filesystem.
+	Cmd *exec.Cmd
+}
+
+// PluginHandshake includes handshake properties for the plugin.
+type PluginHandshake struct {
+	Key   string
+	Value string
+}
+
+// Manager maps the available plugin names to the for different kinds of plugin.Client plugin controllers.
+// T is the custom interface type that the returned GRPCClient implementation implements, e.g. "Differ" for `plugin.Client`s that
+// include a GRPCClient that implements the "Differ" interface:
+// grpcPluginClient, err := c.Client() // Returns a plugin.GRPCClient
+// rawGrpcClientStub, err := grpcPluginClient.Dispense(name) // Calls grpcPluginClient's GRPCClient method and returns the gRPC stub.
+// grpcClient, ok := rawGrpcClient.(Differ) // Asserts the expected type of stub client.
+//
+// The map might include a mapping of "delta" -> plugin.Client to communicate with the Delta plugin.
+type Manager[T any] struct {
+	pluginApplicationClients map[string]*plugin.Client
+}
+
+func NewManager[T any]() *Manager[T] {
+	return &Manager[T]{
+		pluginApplicationClients: make(map[string]*plugin.Client),
+	}
+}
+
+// RegisterPlugin is used to register a new plugin client with the corresponding plugin type.
+func (m *Manager[T]) RegisterPlugin(name string, id PluginIdentity, auth PluginHandshake, p plugin.Plugin) error {
```

Comment:
```suggestion
func (m *Manager[T]) RegisterPlugin(name string, id PluginIdentity, handshake PluginHandshake, p plugin.Plugin) error {
```

It's a handshake not an auth:

* No secrets, can _and should_ log it.
* Comes from regular configuration (e.g. K8s configmap not secret).

Reply:
it kinda is- the plugin client and server agree on a certain key and value that identify one another